WordPress/src/main/java/org/wordpress/android/ui/posts/editor/ImageEditorInitializer.kt

```diff
@@ -70,5 +88,70 @@ class ImageEditorInitializer {
             } else {
                 throw(IllegalArgumentException(IMAGE_STRING_URL_MSG))
             }
+
+        private fun onEditorAction(): (EditorAction) -> Unit = { action ->
+            trackEditorAction(action)
+        }
+
+        private fun trackEditorAction(action: EditorAction) {
+            val stat = when (action) {
+                is EditorShown -> Stat.MEDIA_EDITOR_SHOWN
+                is EditorFinishedEditing -> Stat.MEDIA_EDITOR_USED
+                is EditorCancelled,
+                is PreviewImageSelected,
+                is PreviewInsertImagesClicked,
+                is PreviewCropMenuClicked,
+                is CropOpened,
+                is CropDoneMenuClicked,
+                is CropSuccessful -> null
+            }
+
+            val properties = when (action) {
+                is EditorShown -> mapOf(NUMBER_OF_IMAGES to action.numOfImages)
+                is EditorFinishedEditing -> {
+                    val distinctActions = action.actions.distinct()
+                    if (distinctActions.isNotEmpty()) {
+                        mapOf(ACTIONS to distinctActions.map { it.label })
+                    } else {
+                        null
+                    }
+                }
+                is EditorCancelled,
+                is PreviewImageSelected,
+                is PreviewInsertImagesClicked,
+                is PreviewCropMenuClicked,
+                is CropOpened,
+                is CropDoneMenuClicked,
+                is CropSuccessful -> null
+            }
+
+            val noEditActionPerformed = stat == Stat.MEDIA_EDITOR_USED && properties == null
+            if (stat == null || noEditActionPerformed) {
+                return
+            }
+
+            if (properties == null) {
+                AnalyticsTracker.track(stat)
+            } else {
+                AnalyticsTracker.track(stat, properties)
+            }
+
+            // Also track "media_editor" source with media properties
+            if (action is EditorFinishedEditing) {
+                trackAddMediaFromMediaEditor(action.outputDataList)
+            }
+        }
+
+        private fun trackAddMediaFromMediaEditor(outputDataList: List<OutputData>) {
+            for (outputData in outputDataList) {
+                val properties = AnalyticsUtils.getMediaProperties(
+                    WordPress.getContext(),
+                    false,
+                    Uri.parse(outputData.outputFilePath),
+                    null
+                )
+                properties["via"] = AnalyticsTracker.track(Stat.EDITOR_ADDED_PHOTO_VIA_MEDIA_EDITOR, properties)
```

Comment:
I'm either unfamiliar with this syntax or this code does something else than you intended. I'd expect it saves `Unit` into `properties["via"]` after we track `EDITOR_ADDED_PHOTO_VIA_MEDIA_EDITOR`. Wdyt?

Reply:
My bad, will fix.